src/servers/src/proto.rs

```diff
@@ -390,6 +390,10 @@ impl PromSeriesProcessor {
         let one_sample = series.samples.len() == 1;
 
         for s in series.samples.iter() {
+            // skip NaN value
+            if s.value.is_nan() {
+                continue;
+            }
```

Comment:
Will we output NaN to the result?

Reply:
No, the pipeline cannot handle NaN, so we have to filter it beforehand.